packages/ncids-js/src/components/nci-header/utils/mobile-menu/mobile-menu.ts

```diff
@@ -183,24 +218,40 @@ export class MobileMenu {
 	 * Toggle Menu and Overlay Items
 	 * @private
 	 */
-	private async toggleMenu(): Promise<void> {
+	private async toggleMenu(event: Event): Promise<void> {
```

Comment:
You really should have multiple toggleMenu handlers that do not need the event. Then you should have a shared toggleMenu. This would simplify the poking and prodding at the event below..

Reply:
One more thing when closing the menu, are you setting the menuData back to null?